compiler/luci/service/src/Nodes/CircleRange.cpp

```diff
@@ -24,4 +29,65 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleRange *)
   return _graph->nodes()->create<luci::CircleRange>();
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleRange *node)
+{
+  loco::TensorShape output_shape;
+  output_shape.rank(1);
+
+  auto start_node = dynamic_cast<luci::CircleConst *>(node->start());
+  auto limit_node = dynamic_cast<luci::CircleConst *>(node->limit());
+  auto delta_node = dynamic_cast<luci::CircleConst *>(node->delta());
+
+  if (start_node == nullptr || limit_node == nullptr || delta_node == nullptr)
+  {
+    // We use shape from the node itself
+    loco::TensorShape shape;
+    shape.rank(node->rank());
+    for (uint32_t r = 0; r < node->rank(); ++r)
+    {
+      // Shape inference rules in this file did not consider unknown dimension.
+      // If some node has unknown dimension, 0 is inserted and wrong shape
+      // inference was done as a result.
+      // To fix this, new shape inference algorithm is being implemented.
+      // Until new inference algorithm is fully implemented, unknown dimension
+      // would be represented as 1 along with TFLite expression.
+      shape.dim(r) = node->dim(r).known() ? node->dim(r).value() : 1;
+    }
+    return shape;
```

Comment:
(remark for reviewers) 

This is copy-paste from `own_shape`. This part will be deleted on the next PR.